### ISSUE TITLE
Unable to connect multiple MQTT clients

### ISSUE THREAD

**Issue by ycsin** [zephyrproject-rtos/zephyr#38191]
**Describe the bug**
While I'm testing #38107 with my own application that uses MQTT in one of the thread, I found that only one of the MQTT client is able to connect to my broker. The [mqtt_transport_write](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/mqtt/mqtt.c#L92) throws -12 (ENOMEM) or -116 (ETIMEDOUT) error, which I believe originates from [zsock_send](https://github.com/zephyrproject-rtos/zephyr/blob/main/subsys/net/lib/mqtt/mqtt_transport_socket_tcp.c#L70), anything past here is socket(?) which I don't really understand yet. I'm guessing that the root cause is the same as #38082 but I'm not sure. Does Zephyr support multiple sockets? If it does maybe I'm missing some Kconfigs?

I have tried the followings but the error is still the same.
```c
CONFIG_NET_SOCKETPAIR=y
CONFIG_NET_SOCKETS_NET_MGMT=y
CONFIG_NET_SOCKETS_NET_MGMT_MAX_LISTENERS=3
CONFIG_NET_SOCKETPAIR_BUFFER_SIZE=4096

CONFIG_NET_PPP_RINGBUF_SIZE=1024
CONFIG_NET_PPP_RX_STACK_SIZE=4096
```

**To Reproduce**
Have two MQTT clients trying to connect to the broker in two different threads.

**Expected behavior**
Both MQTT clients work in parallel.

**Impact**
Combined with #38082 this is a very major obstacle for me, if Zephyr isn't able to support multiple sockets in parallel I will have to rethink how I continue with my project, probably involving some major changes to the overall design.

**Logs and console output**
```c
mqtt_transport_write: -12
[00:24:55.820,000] <err> shell_mqtt: mqtt_connect error: -12
[00:24:55.820,000] <wrn> shell_mqtt: Retrying MQTT connect work after 500 ms if network connected
[00:24:55.820,000] <dbg> shell_mqtt.event_handler: Network connected, connecting to MQTT broker in 2 seconds
[00:24:55.820,000] <inf> telemetry_mqtt: MQTT Keepalive
```

**Environment (please complete the following information):**
 - OS: Windows 10
 - Toolchain : gnuarmemb
 - zephyr-v2.6.0-1638-g0296d12675fd

**Comment by ycsin** [comment]
@rlubos @mniestroj @jukkar Any idea?

**Comment by ycsin** [comment]
After getting tips from @mniestroj in Slack, configuring these Kconfigs solve these issues:

```c
CONFIG_POSIX_MAX_FDS=10
CONFIG_NET_BUF_RX_COUNT=32
CONFIG_NET_BUF_TX_COUNT=32
CONFIG_NET_PKT_RX_COUNT=16
CONFIG_NET_PKT_TX_COUNT=16
```

However I only blindly increased these values, so there might be room for optimization.
Using these values also solved #38082
